staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -572,6 +573,34 @@ func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(
 		})
 	}
 
+	if len(instanceMeta.AdditionalLabels) > 0 {
+		klog.V(2).Infof("Adding additional node label(s) from cloud provider: %v", instanceMeta.AdditionalLabels)
+		nodeModifiers = append(nodeModifiers, func(n *v1.Node) {
+			// Store the labels that get overridden for visibility
+			additionalLabelsWithWarnings := make([]string, 0)
+			if n.Labels == nil {
+				n.Labels = map[string]string{}
+			}
+
+			for k, v := range instanceMeta.AdditionalLabels {
+				// Cloud provider should not be using kubernetes.io in labels
+				if isK8s := strings.Contains(k, "kubernetes.io"); isK8s {
```

Comment:
We talked about discarding `kubernetes.io` and `k8s.io` labels. As far as I can tell, [`k8s.io` isn't a format k8s/k8s seems to use, so I didn't bother adding a check](https://github.com/kubernetes/kubernetes/blob/d90342e500a43e5112da04ecefa0c9ae1b1fb6c2/staging/src/k8s.io/api/core/v1/well_known_labels.go#L19-L70). I considered just checking for all of the well known labels, but I didn't see an easily maintainable way to keep the list updated and for the purposes of this new feature, excluding `kubernetes.io` as a substring sounds reasonable.

Reply:
If we do match `[*.]kubernetes.io/*` we should match `[*.]k8s.io/*` as well, even if we avoid using the latter for well-known labels.

> Kubernetes reserves all labels and annotations in the `kubernetes.io` and `k8s.io` namespaces.

and we do have a couple of audit annotations that use the “wrong” (but still protected) convention. Things slip by.

Reply:
Substring is too simple I think though. We'd allow `container-runtime.example/io.kubernetes.iopriority` - unlikely, but not implausible.

Reply:
> Kubernetes reserves all labels and annotations in the kubernetes.io and k8s.io namespaces.

Ah. Good find and tx for sharing. Definitely makes sense to ban both.

> We'd allow container-runtime.example/io.kubernetes.iopriority - unlikely, but not implausible.

Is that a problem? It's not technically the same namespace, right? I don't see a problem with allowing non-standard permutations.

Reply:
A substring match that bans `container-runtime.example/io.kubernetes.iopriority` as a side effect is a bug; we should be matching on the string before the `/` in the label key.

Reply:
Is there a valid case where a k8s/k8s maintained label will not have a `/` in the key? Not that I can see...